src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -589,8 +594,23 @@ InterpreterSelectQuery::InterpreterSelectQuery(
 
 void InterpreterSelectQuery::buildQueryPlan(QueryPlan & query_plan)
 {
+    if (cached_data.contains(query_ptr->getTreeHash()))
```

Comment:
`query_ptr->getTreeHash()` need to be extracted in separate variable, because this method call is heavy.

Reply:
Done.